### ISSUE TITLE
JPG quality keyword in savefig

### ISSUE THREAD

**Issue by fkbreitl** [matplotlib/matplotlib#458]
Hi,

I want to change the JPEG compression ration in savefig.
There seem to be an undocumented extra keyword arguments as pointed at the matplotlib-devel list, but they don't have an effect:

```
         *quality*: The image quality, on a scale from 1 (worst) to
             95 (best). The default is 75. Values above 95 should
             be avoided; 100 completely disables the JPEG
             quantization stage.

         *optimize*: If present, indicates that the encoder should
             make an extra pass over the image in order to select
             optimal encoder settings.

         *progressive*: If present, indicates that this image
             should be stored as a progressive JPEG file.
```

I am using the GTKAgg backend. I also tried the Agg backend but JPG was not supported with this.

I would like to have a proper documentation at http://matplotlib.sourceforge.net/api/pyplot_api.html#matplotlib.pyplot.savefig and an implementation that works.

Cheers

Frank

**Comment by petehuang** [comment]
This issue is stale but updated:

+ 1.3.1 had implemented the `quality` keyword (https://github.com/matplotlib/matplotlib/blob/7131876063f933bc5e5c5d768722dd81bb69038e/lib/matplotlib/backends/backend_gtk.py#L489) but it is currently undocumented

+ The other two keywords are not implemented nor documented

I would tag this with `Documentation` and `rcparams`.

**Comment by tacaswell** [comment]
Passing `quality` to `fig.savefig('/tmp/test80.jpg', quality=80)` does affect the file size and quality in the way I would expect, but it is missing from the savefig docstring.

**Comment by dabana** [comment]
I believe the *progressive* and *optimize* keywords are implemented within the Agg backend through the [Pillow (PIL) JPEG image writer](https://pillow.readthedocs.io/en/3.1.x/handbook/image-file-formats.html) . *FigureCanvasBase* registers Agg as the default backend for saving to JPEG. 

I tested these features on matplotlib version 2.0.2 and they seem to work as expected (at least all of them have an impact on the compressed file size).

I added the two missing keywords to the documentation to *savefig()*. Pull request coming up.
